### PR TITLE
Update to developer guide for the proposed implementation

### PR DESCRIPTION
**Pull request by nobodyishappy**
Added the implementation for grouping to the proposed implementation section of the developer guide.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -158,10 +158,46 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+Step 4. The user executes `addpersontogroup gn/2103T gp/4` command to add a new person 4 to the group named "2103T" from the group list. The `addpersontogroup` command calls `Model#addPersonToGroup(String, Person)`, which finds the group with the same name and add the person to the group.
+
+Step 5. The user executes `removepersonfromgroup gn/2103T gp/4` command to remove person 4 to the group named "2103T" from the group list. The `removepersonfromgroup` command calls `Model#removePersonFromGroup(String, Person)`, which finds the group with the same name and remove the person to the group.
+
+Step 6. The user executes `deletetaskgroup gn/2103T` command to remove a task from the group named "2103T" from the group list. The `deletetaskgroup` command calls `Model#deleteAssignedTaskGroup(String, Task)`, which finds the group with the same name and remove that task from everyone that is in the group.
+
+Step 7. The user executes `deletegroup gn/2103T` command to remove the group from the list. The `deletegroup` command calls `Model#removeGroup(String)`, which finds the group with the same name and remove that group from the list.
+
```
Comment (Kaya3842) [inline docs/DeveloperGuide.md:204]: Maybe you can describe how the UI will show the group(s) a person is currently in, and briefly describe how the UI will change when each command related to grouping is executed.

**Review by Kaya3842 — APPROVED** [review]
More details can be given. But overall it covered the essential parts of an implementation

**Review by Yskie — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -158,10 +158,52 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+New operation are exposed in the `Model` interface are `Model#addPersonToGroup(String, Person)`, `Model#removePersonFromGroup(String, Person)` and `Model#deleteAssignedTaskGroup(String, Task)` which would call `UniquePersonList#add(Person)`, `UniquePersonList#remove(Person)` and `UniquePersonList#deleteAssignedTask(Person)`respectively.
+
+Given below is an example usage scenario and how the grouping mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The list of the `GroupedUniquePersonList` will be empty if there are no groups stored in the storage.
+
+<puml src="diagrams/GroupingState0.puml" alt="GroupingState0" />
+
+Step 2. The user executes `group gn/2103T gp/Ivan gp/Greg gp/Dave` command to group Ivan, Greg and Dave from the displayed person list to one group. The `group` command calls `Model#addGroup(String, List<Person>)`, which creates a new group with that contains the list of people that was indicated by the user.
```
Comment (chin-herng) [inline docs/DeveloperGuide.md:189]: Just a minor typo: `a new group with that contains the list` -> `a new group that contains the list`.

**Review by chin-herng — APPROVED** [review]
LGTM!

**Review by breezetall — APPROVED** [review]
Looks good to me!

**Pull request merged**